webroot/panel/account.php

```diff
@@ -70,6 +70,11 @@
                 }
             }
             break;
+        case "account_deletion_request":
+            if (!$SQL->accDeletionRequestExists($USER->getUID())) {
+                $USER->requestAccountDeletion();
+            }
```

Comment:
This should still check if the user is a member of any PI group before doing this. If they are, you can just `die()` the page, since it's not a state that should be reachable. Frontend verification is ideal, but the backend needs to verify as well since the user can change the frontend and submit the form in a way that you weren't expecting.